include/odp/api/spec/pool_ext_types.h

```diff
@@ -0,0 +1,231 @@
+/* Copyright (c) 2021, Nokia
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier:     BSD-3-Clause
+ */
+
+/**
+ * @file
+ *
+ * Type definitions for external memory pools
+ */
+
+#ifndef ODP_API_SPEC_POOL_EXT_TYPES_H_
+#define ODP_API_SPEC_POOL_EXT_TYPES_H_
+#include <odp/visibility_begin.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <odp/api/std_types.h>
+#include <odp/api/pool.h>
+
+/** @addtogroup odp_pool
+ *  @{
+ */
+
+/**
+ * External memory pool population done
+ *
+ * Application uses this flag to mark the last odp_pool_ext_populate() call, which completes
+ * external memory pool population phase.
+ */
+#define ODP_POOL_POPULATE_DONE 0x1
+
+/**
+ * External memory pool capabilities
+ *
+ * Generic fields (not specific to a pool type) contain capabilities
+ * of the requested pool type.
+ */
+typedef struct odp_pool_ext_capability_t {
+	/** Requested pool type
+	 *
+	 *  Pool type from the odp_pool_ext_capability() call is recorded here for reference. */
+	odp_pool_type_t type;
+
+	/** Maximum number of pools
+	 *
+	 *  Maximum number of external memory pools of the requested type. */
+	uint32_t max_pools;
+
+	/** Minimum size of thread local cache */
+	uint32_t min_cache_size;
+
+	/** Maximum size of thread local cache */
+	uint32_t max_cache_size;
+
+	/** Supported statistics counters */
+	odp_pool_stats_opt_t stats;
+
+	/** Packet pool capabilities  */
+	struct {
+		/** Maximum number of packet buffers */
+		uint32_t max_num_buf;
+
+		/** Maximum packet buffer size in bytes */
+		uint32_t max_buf_size;
+
+		/** ODP header size in bytes
+		 *
+		 *  Application must reserve this many bytes from the start of a packet buffer
+		 *  for ODP implementation usage. When the value is zero, ODP implementation does
+		 *  not need header space to be reserved for it. Application will not modify this
+		 *  memory area (after buffer populate call).
+		 */
+		uint32_t odp_header_size;
+
+		/** ODP trailer size in bytes
+		 *
+		 *  Application must reserve this many bytes from the end of a packet buffer
+		 *  for ODP implementation usage. When the value is zero, ODP implementation does
+		 *  not need trailer space to be reserved for it. Application will not modify this
+		 *  memory area (after buffer populate call).
+		 */
+		uint32_t odp_trailer_size;
+
+		/** Minimum packet buffer pointer alignment in bytes
+		 *
+		 *  Packet buffer pointers populated into a pool must be evenly divisible with
+		 *  this value. The value is 1 when there is no alignment requirement.
+		 */
+		uint32_t min_buf_align;
```

Comment:
I'll add another alignment capability about minimum alignment of the start of the memory area used for an external memory pool. This can e.g. ensure that memory of two pools do not get mapped to the same page.

Reply:
Instead of an alignment capability, I suggest to add capability as "sys_page_sharing".

This capability when **not** set, puts a restriction that the memory area used for two different pools must not have common shared sys pages. 

Memory area of a pool is defined as the memory between the beginning of lowest addressed buffer and the end of the highest addressed buffer populated into the pool.

sys page is defined as any memory area whose address is aligned to and having size of odp_sys_page_size().

This will also avoid the need for having a spec text regarding overlapping memory alloc as it is implied by this.